### ISSUE TITLE
Follow up: Refactoring PR #99

### ISSUE THREAD

**Issue by thetechnick** [kubermatic/kubecarrier#107]
This is a follow up issue to https://github.com/kubermatic/kubecarrier/pull/99

I like the simplification of the API and having proper tests, but I have some issues with this PR:

Using string replacements instead of writing a patch for Kustomize is IMHO completely defeating the purpose of having Kustomize in the first place.
The big plus of Kustomize is (for me at least
a) we can work in a structured way with the objects, without having to worry about side effects
b) we can reuse the code-generators from kubebuilder

moving the kustomize package from /pkg/internal/kustomize to /pkg/internal/resources/internal/kustomize is not only redundand (double internal in path), it also is an unnecessarily deep folder path.

TL;DR: I want the kustomize package back to /pkg/internal/kustomize and overrides via the kustomization.yaml

_Originally posted by @thetechnick in https://github.com/kubermatic/kubecarrier/pull/99#issuecomment-571479493_
